docs/DeveloperGuide.md

```diff
@@ -6,17 +6,146 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Our program is structured as pictured in the following class diagram:
+![ClassDiagram.png](diagrams/ClassDiagram.png)
+
+All the inheritors of Command are features that will be described below.
+
+### Addition feature
+
+The addition mechanism is facilitated by 'Parser', 'Add', 'ModuleList' and 'UI', where a Module object will be created according to the content inputted by the user and stored in the module list.
```

Comment:
Can consider using 'code' feature to better distinguish the class and to draw attention. It  will look something like `Parser`.